Formula/ghc@8.6.rb

```diff
@@ -52,31 +60,46 @@ def install
     ENV["LD"] = "ld"
     ENV["PYTHON"] = Formula["python@3.9"].opt_bin/"python3"
 
-    # Build a static gmp rather than in-tree gmp, otherwise all ghc-compiled
-    # executables link to Homebrew's GMP.
-    gmp = libexec/"integer-gmp"
-
-    # GMP *does not* use PIC by default without shared libs so --with-pic
-    # is mandatory or else you'll get "illegal text relocs" errors.
-    resource("gmp").stage do
-      system "./configure", "--prefix=#{gmp}", "--with-pic", "--disable-shared",
-                            "--build=#{Hardware.oldest_cpu}-apple-darwin#{OS.kernel_version.major}"
-      system "make"
-      system "make", "install"
+    args = %w[--enable-numa=no]
+    on_macos do
+      # Build a static gmp rather than in-tree gmp, otherwise all ghc-compiled
+      # executables link to Homebrew's GMP.
+      gmp = libexec/"integer-gmp"
+
+      # GMP *does not* use PIC by default without shared libs so --with-pic
+      # is mandatory or else you'll get "illegal text relocs" errors.
+      resource("gmp").stage do
+        system "./configure", "--prefix=#{gmp}", "--with-pic", "--disable-shared",
+                              "--build=#{Hardware.oldest_cpu}-apple-darwin#{OS.kernel_version.major}"
+        system "make"
+        system "make", "install"
+      end
+
+      args = ["--with-gmp-includes=#{gmp}/include",
+              "--with-gmp-libraries=#{gmp}/lib"]
```

Comment:
I'm wondering if we even need this any of our GHC formulae.  We might want to check in a follow up PR if the `build.mk` change I made and `--with-intree-gmp` in the newer GHC formulae is sufficient to statically link `gmp` on macOS as well.  If so, we'd be able to reduce the difference between macOS and Linux in the GHC formulae to just using different bootstrap binaries.

Reply:
Yea, I suspect we should be able to do the same thing for macOS that we currently do on Linux. That's been on my to-do list for a while, but haven't found the time to do it yet.